### PR TITLE
Ecr lambda permissions

### PR DESCRIPTION
**Pull request by ian-hoyle**
lots of changes when I run terraform plan on intg not associated with is branch
 Plan: 4 to add, 43 to change, 4 to destroy.
 are they ok to ignore
 
 including
 module.cloudfront_certificate.aws_acm_certificate.cert 
 module.consignment_api_certificate.aws_acm_certificate.cert
 module.frontend.aws_elasticache_replication_group.redis_replication_group will be updated in-place

### CODE REVIEW

File: root_draft_metadata.tf [hunk 1]
```diff
@@ -2,17 +2,19 @@ module "draft_metadata_validator_lambda" {
...
-      environment    = local.environment
-      parameter_name = local.keycloak_backend_checks_secret_name
-      bucket_name    = local.draft_metadata_s3_bucket_name
-      kms_key_arn    = module.s3_internal_kms_key.kms_key_arn
+      account_id            = var.tdr_account_number
+      environment           = local.environment
+      parameter_name        = local.keycloak_backend_checks_secret_name
+      bucket_name           = local.draft_metadata_s3_bucket_name
+      kms_key_arn           = module.s3_internal_kms_key.kms_key_arn
+      management_account_id = local.ecr_account_number
```
Comment (TomJKing) [inline root_draft_metadata.tf:17]: For naming consistency can you change the name of this variable from `management_account_id` to `ecr_account_number` to make it clear what it is.

File: templates/iam_policy/draft_metadata_validator_lambda.json.tpl [hunk 2]
```diff
@@ -38,6 +38,14 @@
...
       ],
       "Resource": "${kms_key_arn}"
+    },
+    {
+        "Effect": "Allow",
+        "Action": [
+         "ecr:BatchGetImage",
+         "ecr:GetDownloadUrlForLayer"
+     ],
+     "Resource": "arn:aws:ecr:eu-west-2:${management_account_id}:repository/draft-metadata-validator"
```
Comment (TomJKing) [inline templates/iam_policy/draft_metadata_validator_lambda.json.tpl:48]: See comment above about changing the account id variable name to `ecr_account_id` for naming consistency

File: root_draft_metadata.tf [hunk 3]
```diff
@@ -21,9 +23,6 @@ module "draft_metadata_validator_lambda" {
     CLIENT_SECRET_PATH = local.keycloak_backend_checks_secret_name
     BUCKET_NAME        = local.draft_metadata_s3_bucket_name
   }
-  lambda_invoke_permissions = {
```
Comment (TomJKing) [inline root_draft_metadata.tf:24]: Thanks for deleting this.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments. Thanks

**Comment by TomJKing** [comment]
> _... lots of changes when I run terraform plan on intg not associated with is branch..._

@ian-hoyle This should be resolved now. This happens when a newer version of Terraform is used for `plan` or `apply` operations.

File: root_draft_metadata.tf (hunk 1 above)
Comment (ian-hoyle) [reply root_draft_metadata.tf]: done

File: templates/iam_policy/draft_metadata_validator_lambda.json.tpl (hunk 2 above)
Comment (ian-hoyle) [reply templates/iam_policy/draft_metadata_validator_lambda.json.tpl]: done

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.

**Review by annielh — APPROVED** [review]

**Pull request merged**